eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -1205,21 +1207,41 @@ object Router {
    * Have to split ids because otherwise message could be too big
    * there could be several reply_channel_range messages for a single query
    */
-  def split(shortChannelIds: SortedSet[ShortChannelId], channelRangeChunkSize: Int): List[ShortChannelIdsChunk] = {
-    // this algorithm can split blocks (meaning that we can in theory generate several replies with the same first_block/num_blocks
-    // and a different set of short_channel_ids) but it doesn't matter
-    if (shortChannelIds.isEmpty) {
-      List(ShortChannelIdsChunk(0, 0, List.empty))
+  def split(shortChannelIds: SortedSet[ShortChannelId], firstBlockNum: Long, numberOfBlocks: Long, channelRangeChunkSize: Int): List[ShortChannelIdsChunk] = {
+    val it = shortChannelIds.iterator.dropWhile(_.blockHeight < firstBlockNum).takeWhile(_.blockHeight < firstBlockNum + numberOfBlocks)
```

Comment:
I like including RFC snippets for that kind of precise things, saves time for reviewers.

```suggestion
    // BOLT 7: MUST encode a short_channel_id [...] in blocks first_blocknum to first_blocknum + number_of_blocks - 1
    val it = shortChannelIds.iterator.dropWhile(_.blockHeight < firstBlockNum).takeWhile(_.blockHeight < firstBlockNum + numberOfBlocks)
```